cmd/nonmem.go

```diff
@@ -0,0 +1,395 @@
+
+	return nmExecutable + " " + strings.Join(cmdArgs, " ")
+}
+
+//modelName is the full file + ext representation of the model (ie acop.mod)
+//directory is the directory in which we will be performing cleanup
+//exceptions is a variadic input for allowing exceptions / overrides.
+//We're taking a local model because grid engine execution doesn't wait for completion. Nothing to do :)
+func filesToCleanup(model NonMemModel, exceptions ...string) runner.FileCleanInstruction {
+	fci := runner.FileCleanInstruction{
+		Location: model.OutputDir,
+	}
+
+	files := getActionableFileList(model.FileName, model.Settings.CleanLvl, model.OutputDir)
+
+	for _, v := range files {
+		if !isFilenameInExceptions(exceptions, v) {
+			//Let's add it to the cleanup list if it's not in the exclusions
+			fci.FilesToRemove = append(fci.FilesToRemove, newTargetFile(v, model.Settings.CleanLvl))
+		}
+	}
+
+	return fci
+}
+
+func isFilenameInExceptions(haystack []string, needle string) bool {
+
+	for _, v := range haystack {
+		if v == needle {
+			return true
+		}
+	}
+
+	//No matches
+	return false
+}
+
+func newTargetFile(filename string, level int) runner.TargetedFile {
+	return runner.TargetedFile{
+		File:  filename,
+		Level: level,
+	}
+}
+
+func filesToCopy(model NonMemModel, mandatoryFiles ...string) runner.FileCopyInstruction {
+	fci := runner.FileCopyInstruction{
+		CopyTo:   model.OriginalPath,
+		CopyFrom: model.OutputDir,
+	}
+
+	//Process mandatory files first
+	for _, v := range mandatoryFiles {
+		//Crank it up to 11
+		fci.FilesToCopy = append(fci.FilesToCopy, newTargetFile(v, 11))
+	}
+
+	//Create Target File Entries
+	for _, v := range getActionableFileList(model.FileName, model.Settings.CopyLvl, model.OutputDir) {
+		fci.FilesToCopy = append(fci.FilesToCopy, newTargetFile(v, model.Settings.CopyLvl))
+	}
+
+	return fci
+}
+
+func getActionableFileList(filename string, level int, filepath string) []string {
+	var output []string
+	files := make(map[int][]string)
+
+	//These files are files that may be desired above the normal.
+	files[2] = []string{
+		"background.set",
+		"compile.lnk",
+		"FCON",
+		"FDATA",
+		"FMSG",
+		"FREPORT",
+		"FSIZES",
+		"FSTREAM",
+		"FSUBS",
+		"FSUBS.0",
+		"FSUBS.o",
+		"FSUBS_MU.F90",
+		"FSUBS.f90",
+		"fsubs.f90",
+		"FSUBS2",
+		"gfortran.txt",
+		"GFCOMPILE.BAT",
+		"INTER",
+		"licfile.set",
+		"linkc.lnk",
+		"LINK.LNK",
+		"LINKC.LNK",
+		"locfile.set",
+		"maxlim.set",
+		"newline",
+		"nmexec.set",
+		"nmpathlist.txt",
+		"nmprd4p.mod",
+		"nobuild.set",
+		"parafile.set",
+		"parafprint.set",
+		"prcompile.set",
+		"prdefault.set",
+		"prsame.set",
+		"PRSIZES.f90",
+		"rundir.set",
+		"runpdir.set",
+		"simparon.set",
+		"temp_dir",
+		"tprdefault.set",
+		"trskip.set",
+		"worker.set",
+		"xmloff.set",
+		"fort.2001",
+		"fort.2002",
+		"flushtime.set",
+	}
+
+	//Get the files from the model
+	//Look specifically for .mod file
+	fileContents, err := ioutil.ReadFile(path.Join(filepath, filename+".mod"))
```

Comment:
we need to actually expand any places that we look for `.mod` to also look for `.ctl` --> .mod = psn convention, .ctl is general nonmem convention metrum uses

Reply:
That makes sense. I can just iterate through "expected" extensions to find any files

Reply:
So what makes even more sense is to actually just pass down the full file (name+ext) to this method and extract the filename from it. This way we don't have to curse through a list of "known" extensions. It's just whatever the user specified in the args.